### ISSUE TITLE
const dir = t.testDir()

### ISSUE THREAD

**Issue by isaacs** [tapjs/tapjs#595]
I find myself doing this in a lot of my tests:

```js
const t = require('tap')
const {resolve, basename} = require('path')
const me = resolve(__dirname, basename(__filename, '.js'))
const rimraf = require('rimraf')
const mkdirp = require('mkdirp')
rimraf.sync(me)
mkdirp.sync(me)
t.teardown(() => rimraf.sync(me))
```

It'd be nice if there was a way to just do:

```js
const me = t.testdir()
```

Then when the test tears down, remove it, unless an environment variable (which could be set via a tap runtime flag) was set to tell it not to.

**Comment by isaacs** [comment]
For bonus points, maybe it could also do a Tacks-style fixture directory setup to populate the contents of the test directory:

```js
const dir = t.testdir({
  'package.json': JSON.stringify({name: 'foo'}),
  lib: {
    'foo.js': 'console.log("foo")',
    'link.js': t.fixture('symlink', 'foo.js'),
    'hardlink.js': t.fixture('link', 'foo.js'),
  },
})
```

**Comment by isaacs** [comment]
Landed in 14.7.0
